include/uam/uam_driver_ros.h

```diff
@@ -220,6 +249,110 @@ class UamROS
    */
   void updateStatus(const protocol::sensing_data::SensingDataHeader& sensing_data, const bool override_check = false);
 
+  /**
+   * @brief Advertise safety area type publishers
+   * @param[in] publishers - List of publishers to advertise
+   */
+  void advertiseSafetyAreaPublishers();
+
+  /**
+   * @brief Publish Safety Area scans
+   * @param[in] area_number - The lidar safety area number
+   */
+  void publishSafetyArea(const size_t area_number);
+
+  /**
+   * @brief Read Safety areas from lidar
+   */
+  void readSafetyAreas();
+
+  /**
+   * @brief Publish safety area violation markers
+   * @param[in] packet - Incoming packet
+   * @param[in] safety_area - Safety area to validate the ranges against
+   */
+  template <typename T>
+  void publishWarningMarkers(const T& packet, const sensor_msgs::LaserScan& safety_area)
+  {
+    auto getLut = [this](const ScanParameters& scan_params, const T& packet)
+    {
+      LaserScanCachedLookup lut;
+      lut.scan_angle_min = scan_params_.getAngleMinLimit();
+      lut.scan_angle_max = scan_params_.getAngleMaxLimit();
+      lut.scan_angle_increment = scan_params_.getAngleIncrement();
+
+      lut.sin_lookup.clear();
+      lut.cos_lookup.clear();
+
+      lut.sin_lookup.reserve(packet.ranges.size());
+      lut.cos_lookup.reserve(packet.ranges.size());
+
+      for (size_t i = 0; i < packet.ranges.size(); ++i)
```

Comment:
Would there every be a time where `scan_params_.getAngleMaxLimit()` and `scan_params_.getAngleMinLimit() + scan_params_.getAngleIncrement() * (packet.ranges.size() - 1)` would every be different? MaxAngle and (AngleIncrement, Count) are redundant in some sense. Just curious if anything needs to care.